docs/src/main/asciidoc/security-oidc-auth0-tutorial.adoc

```diff
@@ -0,0 +1,771 @@
+[id="security-oidc-auth0-tutorial"]
+= Protect Quarkus web application by using `Auth0` OpenID Connect provider
+include::_attributes.adoc[]
+:diataxis-type: tutorial
+:categories: security,web
+
+xref:security-architecture.adoc[Quarkus Security] provides comprehensive OpenId Connect (OIDC) and OAuth2 support with its `quarkus-oidc` extension, supporting both xref:security-oidc-code-flow-authentication.adoc[Authorization code flow] and xref:security-oidc-bearer-token-authentication.adoc[Bearer token] authentication mechanisms.
+
+With Quarkus, you can easily configure OIDC providers such as link:https://www.keycloak.org/documentation[Keycloak], link:https://developer.okta.com/[Okta], link:https://auth0.com/docs/[Auth0], and other xref:security-openid-connect-providers.adoc[well-known social OIDC and OAuth2 providers].
+
+Learn how to use the Quarkus OpenID Connect extension (`quarkus-oidc`) together with the https://auth0.com/docs/[Auth0] OIDC provider to protect your API endpoints.
+
+== Prerequisites
+
+Reviewing https://auth0.com/docs/[Auth0 documentation] is recommended.
+
+Reviewing Quarkus OIDC documents about xref:security-oidc-code-flow-authentication.adoc[Authorization code flow] and xref:security-oidc-bearer-token-authentication.adoc[Bearer token] authentication mechanisms will make it easier for you to complete this tutorial.
+
+:sectnums:
+:sectnumlevels: 3
+
+== Create an Auth0 application
+
+. Go to the `Auth0` dashboard and create a regular web application. For example, create an `Auth0` application called `QuarkusAuth0`.
+
+image::auth0-create-application.png[Create Auth0 application]
+
+.Result
+Your`Auth0` application gets created with a client ID, secret, and HTTPS-based domain.
```

Comment:
Does Auth0 need to be inside backticks?
It's a valid proper noun. If Vale rejects it, I can fix that rule to remove the false positive.

```suggestion
Your Auth0 application gets created with a client ID, secret, and HTTPS-based domain.
```